lib/src/lifecycle_module.dart

```diff
@@ -237,6 +237,13 @@ abstract class LifecycleModule implements DisposableManager {
     if (_childModules.contains(newModule)) {
       return new Future(() {});
     }
+
+    if (isUnloaded || isUnloading) {
+      final state = isUnloaded ? 'unloaded' : 'unloading';
```

Comment:
There was some discussion in the Dart chat about defaulting to `var` again unless there is a particular reason to use `final` now that the analyzer can detect type-unstable reassignments.